torch/_inductor/sizevars.py

```diff
@@ -243,46 +225,86 @@ def guard_equals(self, left: Expr, right: Expr) -> Expr:
         assert self.shape_env.evaluate_expr(sympy.Eq(left, right))
         return left
 
-    # See Note [maybe_guard vs guard]
-    def maybe_guard_equals(self, left: Expr, right: Expr) -> bool:
-        """if left==right, guard on that fact and return true"""
+    # Note - [On Should Optimize]
+    #
+    # The should_optimize_* family of functions below replaces a prior system, called maybe_guard_*. The prior system
+    # operated by providing esentially a question, where the size hinted values were evaluted. If the condition was
+    # true, we add a guard and return True, otherwise, False.
+    #
+    # def maybe_guard_foo(args):
+    #   if size_hinted_check(args):
+    #       return False # No guard, no optim
+    #   guard(args) # Make a guard
+    #   return True # Safe to apply optimization
+    #
+    # The prior system incurred a guard, and green lit an optimization.
+    #
+    # The new system works in reverse - in the new system, if we know that the inputs are static, and evaluate the
+    # condition as true, we green light the optimization, and we do not incur a guard. If we cannot prove that, we
+    # return False.
+    #
+    # def maybe_guard_foo(args):
+    #   if all_static(args):
+    #       return True # Safe to apply optimization
+    #   else:
+    #       return False # No guard, no optim
+
+    # See Note - [On Should Optimize]
+    def should_optimize_equals(self, left: Expr, right: Expr) -> bool:
```

Comment:
bikeshedding the name, how about `statically_known_equals`? `should_optimize_equals` doesn't really tell me what the meaning of True/False from the method is (I happen to know, because I know the impl, but just looking at the function name it's unclear.)